File: tools/terminal_ops/terminal_entry_audit.mjs
#!/usr/bin/env node
import fs from 'fs';
import path from 'path';

const filePath = process.argv[2] || 'terminal/index.html';
const html = fs.readFileSync(filePath, 'utf8');

const scriptMatches = [...html.matchAll(/<script[^>]+src="([^"]+)"/g)].map(m => m[1]);
const cssMatches = [...html.matchAll(/<link[^>]+href="([^"]+)"/g)].map(m => m[1]);
const uiV3Refs = scriptMatches.filter(s => s.includes('ui-v3/'));

const duplicateWarnings = [];
const ownershipHints = new Map();
for (const s of scriptMatches) {
  const key = path.basename(s);
  if (ownershipHints.has(key)) duplicateWarnings.push(`duplicate script basename: ${key}`);
  ownershipHints.set(key, s);
}

console.log(JSON.stringify({
  file: filePath,
  css: cssMatches,
  jsEntrypoints: scriptMatches,
  uiV3Refs,
  duplicateWarnings,
}, null, 2));

File: tools/terminal_ops/terminal_entry_audit.py
from bs4 import BeautifulSoup


def audit_index(path="terminal/index.html"):
    with open(path, "r", encoding="utf-8") as f:
        html = f.read()

    soup = BeautifulSoup(html, "html.parser")

    scripts = [s.get("src") for s in soup.find_all("script") if s.get("src")]
    css = [l.get("href") for l in soup.find_all("link") if l.get("href")]

    print("JS entrypoints:", scripts)
    print("CSS:", css)


if __name__ == "__main__":
    audit_index()
